### PR TITLE
Prevent LineStrings to be filled

### PR DESCRIPTION
**Pull request by Jenselme**
Lines imported from KML have a filled style which makes them look as polygon once printed. This PR solves this problem.

However, as for #239, it may be a bug in openlayer. @elemoine any opinion?

### CODE REVIEW

File: src/services/print.js [hunk 1]
```diff
@@ -364,8 +364,15 @@ ngeo.Print.prototype.encodeVectorLayer_ = function(arr, layer, resolution) {
         var style = styles[j];
         var styleId = goog.getUid(style).toString();
         var featureStyleProp = ngeo.Print.FEAT_STYLE_PROP_PREFIX_ + j;
+        var geometryType;
+        if (goog.isDef(geojsonFeature.geometry) &&
+                goog.isDef(geojsonFeature.geometry.type)) {
+          geometryType = geojsonFeature.geometry.type;
+        } else {
+          geometryType = 'FeatureCollection';
+        }
```
Comment (elemoine) [inline src/services/print.js]: Can't you just use `feature.getGeometry().getType()`?

File: src/services/print.js [hunk 2]
```diff
@@ -408,7 +416,7 @@ ngeo.Print.prototype.encodeVectorStyle_ =
   var imageStyle = style.getImage();
   var strokeStyle = style.getStroke();
   var textStyle = style.getText();
-  if (!goog.isNull(fillStyle)) {
+  if (!goog.isNull(fillStyle) && geometryType !== 'LineString') {
```
Comment (elemoine) [inline src/services/print.js]: What about MultiLineString?

**Comment by elemoine** [comment]
Fixing/working around this problem in the ngeo print service makes sense to me.

File: src/services/print.js (hunk 2 above)
Comment (Jenselme) [inline src/services/print.js]: I forgot them.

File: src/services/print.js (hunk 1 above)
Comment (Jenselme) [inline src/services/print.js]: I can, thanks

**Comment by Jenselme** [comment]
PR updated, commits squashed.

File: src/services/print.js [hunk 3]
```diff
@@ -364,8 +364,9 @@ ngeo.Print.prototype.encodeVectorLayer_ = function(arr, layer, resolution) {
         var style = styles[j];
         var styleId = goog.getUid(style).toString();
         var featureStyleProp = ngeo.Print.FEAT_STYLE_PROP_PREFIX_ + j;
+        var geometryType = feature.getGeometry().getType();
```
Comment (elemoine) [inline src/services/print.js]: You just need to call `getGeometry().getType()` once for each feature. And you also want to deal with the case where the feature does not have a geometry (`getGeometry` returns `null` or `undefined`).

File: src/services/print.js [hunk 4]
```diff
@@ -408,7 +410,9 @@ ngeo.Print.prototype.encodeVectorStyle_ =
   var imageStyle = style.getImage();
   var strokeStyle = style.getStroke();
   var textStyle = style.getText();
-  if (!goog.isNull(fillStyle)) {
+  var isLine = geometryType === 'LineString' ||
+          geometryType === 'MultiLineString';
```
Comment (elemoine) [inline src/services/print.js]: Please use `ol.geom.GeometryType.LINE_STRING` and `ol.geom.GeometryType.MULTI_LINE_STRING`. And add `goog.include('ol.geom.GeometryType')`.

**Comment by elemoine** [comment]
Sorry, I still have comments.

**Comment by Jenselme** [comment]
> Sorry, I still have comments.

No problem. PR updated.

File: src/services/print.js [hunk 5]
```diff
@@ -360,12 +361,19 @@ ngeo.Print.prototype.encodeVectorLayer_ = function(arr, layer, resolution) {
       if (goog.isNull(geojsonFeature.properties)) {
         geojsonFeature.properties = {};
       }
+      var geometry = feature.getGeometry();
+      var geometryType;
+      if (goog.isDef(geometry) && !goog.isNull(geometry)) {
+        geometryType = geometry.getType();
+      } else {
+        geometryType = '';
+      }
```
Comment (elemoine) [inline src/services/print.js]: I think we can just skip the feature if it doesn't have a geometry. And note that you can use `goog.isDefAndNotNull`.

Comment (Jenselme) [inline src/services/print.js]: > And note that you can use goog.isDefAndNotNull.

Noted.

> I think we can just skip the feature if it doesn't have a geometry

What do you mean by that?

Comment (elemoine) [inline src/services/print.js]: In the loop over the features we can just skip the feature (`continue`) if the feature does not have a geometry.

**Comment by Jenselme** [comment]
PR updated

**Comment by elemoine** [comment]
Still not exactly to my taste :) But I'll merge this and slightly change the code with a separate commit/PR.

Thanks for your contribution. (It may be time to create a contributors file and add your name there.)

**Comment by Jenselme** [comment]
> It may be time to create a contributors file and add your name there.

This up to you I guess

**Pull request merged**